toucan_connectors/mongo/mongo_connector.py

```diff
@@ -259,15 +260,38 @@ def get_df_with_regex(
         # Since Mongo '$regex' operator doesn't work with integer values, we need to check the stringified versions
         regex_step = {
             '$expr': {
-                '$regexMatch': {
-                    'input': {'$toString': f'${field}'},
-                    'regex': regex.pattern,
-                    'options': 'i',  # i -> Case insensitivity
-                }
+                '$or': [
+                    {
+                        '$regexMatch': {
+                            'input': {'$toString': f'${field}'},
+                            'regex': regex.pattern,
+                            'options': 'i',  # i -> Case insensitivity
+                        }
+                    }
+                    for field in fields
+                ]
             }
         }
         data_source.query.append({'$match': regex_step})
-        return self.get_slice(data_source, permissions, limit=limit).df
+        return self.get_slice(data_source, permissions, limit=limit, offset=offset)
+
+    def get_df_with_regex(
+        self,
+        data_source: MongoDataSource,
+        field: str,
```

Comment:
Why not support list here too? I don't understand.

Reply:
Because it wasn't supported in the original source code so we kept the function's contract